### PR TITLE
Adding user API

### PR DESCRIPTION
**Pull request by porcellus**
## Summary of change
Adding user GET/PUT API

## Related issues
- #322 
- #333 

## Test Plan
Added some of the basic testcases the rest will be collected on the passwordless->version branch PR.

### CODE REVIEW

File: src/main/java/io/supertokens/webserver/api/passwordless/UserAPI.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            } else if (email != null) {
+                user = Passwordless.getUserByEmail(main, Utils.normaliseEmail(email));
+            } else {
+                user = Passwordless.getUserByPhoneNumber(main, phoneNumber);
+            }
+
+            if (user == null) {
+                JsonObject result = new JsonObject();
+                result.addProperty("status", userId != null ? "UNKNOWN_USER_ID_ERROR"
+                        : email != null ? "UNKNOWN_EMAIL_ERROR" : "UNKNOWN_PHONE_NUMBER_ERROR");
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/webserver/api/passwordless/UserAPI.java:80]: Add brackets like `userId != null ? "UNKNOWN_USER_ID_ERROR" : (email != null ? "UNKNOWN_EMAIL_ERROR" : "UNKNOWN_PHONE_NUMBER_ERROR")` just to make this super clear

File: src/main/java/io/supertokens/passwordless/Passwordless.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                throw (DuplicateEmailException) e.actualException;
+            }
+
+            if (e.actualException instanceof DuplicatePhoneNumberException) {
+                throw (DuplicatePhoneNumberException) e.actualException;
+            }
+        }
+    }
+
+    public static class FieldUpdate {
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/passwordless/Passwordless.java:100]: Should write comments about the purpose of this.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/main/java/io/supertokens/webserver/api/passwordless/UserAPI.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        String userId = InputParser.parseStringOrThrowError(input, "userId", false);
+
+        FieldUpdate emailUpdate = !input.has("email") ? null
+                : new FieldUpdate(input.get("email").isJsonNull() ? null
+                        : Utils.normaliseEmail(InputParser.parseStringOrThrowError(input, "email", false)));
+
+        FieldUpdate phoneNumberUpdate = !input.has("phoneNumber") ? null
+                : new FieldUpdate(input.get("phoneNumber").isJsonNull() ? null
+                        : InputParser.parseStringOrThrowError(input, "phoneNumber", false));
+
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/webserver/api/passwordless/UserAPI.java:108]: If someone doesn't pass email and phoneNumber, then we must throw a bad request?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/main/java/io/supertokens/webserver/api/passwordless/UserAPI.java (hunk 3 above)
Comment (porcellus) [reply src/main/java/io/supertokens/webserver/api/passwordless/UserAPI.java:108]: It's a no-op, so I'm not sure if we need to throw: the request is not to update either, so we don't.

Comment (rishabhpoddar) [reply src/main/java/io/supertokens/webserver/api/passwordless/UserAPI.java:108]: Sounds good.

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**